nano/node/node.cpp

```diff
@@ -2906,6 +2907,28 @@ void nano::active_transactions::request_confirm (std::unique_lock<std::mutex> &
 					}
 					election_l->update_dependent ();
 				}
+				// Recalculate work for local blocks if difficulty is below active_difficulty
+				if (election_l->local_block && !election_l->ongoing_work_update)
+				{
+					auto block_l (election_l->status.winner);
+					uint64_t difficulty (0);
+					nano::work_validate (*block_l, &difficulty);
+					if (difficulty < active_difficulty)
+					{
+						auto node_l (node.shared ());
+						auto callback = [node_l, block_l, election_l](boost::optional<uint64_t> const & work_a) {
+							if (work_a && !nano::work_validate (block_l->root (), work_a.get ()))
+							{
+								block_l->block_work_set (work_a.get ());
+								node_l->active.update_difficulty (*block_l);
+								node_l->network.flood_block (block_l);
+								election_l->ongoing_work_update = false;
+							}
+						};
+						node.work_generate (block_l->root (), callback, difficulty);
+						election_l->ongoing_work_update = true;
```

Comment:
It seems possible (although very unlikely) that the work is calculated and the callback gets called and sets `election_l->ongoing_work_update = false;` before `election_l->ongoing_work_update = true;` is set as they will be on different threads. Will that cause any issues?

Reply:
it can be set after diff is checked against active_difficulty

Reply:
looks good